src/lib.rs

```diff
@@ -90,10 +91,39 @@ struct InnerEvent {
 
 impl InnerEvent {
     fn new(event: Event, api_key: String) -> Self {
+        let mut properties = event.properties;
+
+        // Add $lib_name and $lib_version to the properties
+        properties.props.insert(
+            "$lib_name".into(),
+            serde_json::Value::String("posthog-rs".into()),
+        );
+
+        let version_str = env!("CARGO_PKG_VERSION");
+        properties.props.insert(
+            "$lib_version".into(),
+            serde_json::Value::String(version_str.into()),
+        );
+
+        if let Ok(version) = version_str.parse::<Version>() {
+            properties.props.insert(
+                "$lib_version__major".into(),
```

Comment:
the double underscore is intentional, see https://us.posthog.com/project/2/data-management/properties/01924912-0dda-7ac0-bfed-a5d18e8392f3
<img width="452" alt="Screenshot 2025-01-06 at 15 09 19" src="https://github.com/user-attachments/assets/5a1f2b26-cb23-42b8-97e7-3ea422afd4de" />